src/bicep-types/src/writers/markdown.ts

```diff
@@ -268,6 +268,49 @@ export function writeMarkdown(types: BicepType[], fileHeading?: string) {
     }
   }
 
+  function isComplexType(type: BicepType) {
+    switch (type.type) {
+      case TypeBaseKind.ResourceType:
+      case TypeBaseKind.ResourceFunctionType:
+      case TypeBaseKind.ObjectType:
+      case TypeBaseKind.DiscriminatedObjectType:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  function getComplexTypeName(type: BicepType) {
+    switch (type.type) {
+      case TypeBaseKind.ResourceType:
+        return (type as ResourceType).name;
+      case TypeBaseKind.ResourceFunctionType:
+        return (type as ResourceFunctionType).name;
+      case TypeBaseKind.ObjectType:
+        return (type as ObjectType).name;
+      case TypeBaseKind.DiscriminatedObjectType:
+        return (type as DiscriminatedObjectType).name;
+      default:
+        return undefined;
+    }
+  }
+
+  function dedupeAndFilter(typesToWrite: BicepType[]) {
+    const uniqueTypesMap = new Map<string, BicepType>();
+
+    for (const type of typesToWrite) {
+      if (isComplexType(type)) {
+        const typeName = getComplexTypeName(type);
+
+        if (typeName && !uniqueTypesMap.has(typeName)) {
+          uniqueTypesMap.set(typeName, type);
+        }
+      }
+    }
+
+    return Array.from(uniqueTypesMap.values());
+  }
```

Comment:
Couldn't we instead do:
```suggestion
  function dedupeAndFilter(typesToWrite: BicepType[]) {
    return [...new Set(typesToWrite)];
  }
```

Reply:
Updated to dedupe by object reference.